### ISSUE TITLE
bug: registry assets fail to load in prod build

### ISSUE THREAD

**Issue by conorsch** [penumbra-zone/dex-explorer#121]
Spent some time debugging a disparity between the deployed testnet-facing instance (https://dex-explorer.testnet.plinfra.net) and the local dev env. That investigation resulted in https://github.com/penumbra-zone/dex-explorer/pull/120, which is good progress, but we're still not quite there: the v2 version now deployed doesn't load asset info correctly.

When running the prod build, I see in the browser network console a 404 when accessing this URL: `https://raw.githubusercontent.com/prax-wallet/registry/main/registry/chains/.json` That's obviously a concatenation error where the env var `PENUMBRA_CHAIN_ID=penumbra-testnet-phobos-2` did not get interpolated into the string. It's definitely set, though, and the `pnpm dev` environment loads it just fine. Why isn't the chain id loading in the prod build?

Previously we were using the bundled version of the assets (https://github.com/penumbra-zone/dex-explorer/issues/57#issuecomment-2363412022), but I believe recent work on the v2 interface moved that to remote fetching.

**Comment by grod220** [comment]
Some ideas, but happy to pair on this though!

Wonder if we need to read from the build args and insert as env vars in the [containerfile](https://github.com/penumbra-zone/dex-explorer/blob/main/Containerfile):

```
ARG PENUMBRA_GRPC_ENDPOINT
ENV PENUMBRA_GRPC_ENDPOINT=${PENUMBRA_GRPC_ENDPOINT}
```

Do we need to pass in secrets within the [deploy action](https://github.com/penumbra-zone/dex-explorer/blob/main/.github/workflows/deploy.yml)?

Another idea entirely is to write a `env.production` file with all the secrets ([docs](https://nextjs.org/docs/pages/building-your-application/configuring/environment-variables#default-environment-variables)). Frontenders are using `env.development` locally and nextjs automatically loads the variables.

**Comment by conorsch** [comment]
Ah, are you suggesting that the the code has regressed and we lost the ability to configure the asset registry info at runtime? In #73 we discussed the requirement to support multiple chains via our published image:

> We currently build a container image, ghcr.io/penumbra-zone/dex-explorer, to make it easy for anyone to run a dex-explorer. There are currently multiple deployments, tracking different chains, based on this image. We must update the application config such that the same built artifact, i.e. the container, can be used to run against multiple chains.

For example, RC runs an instance and they need to be able to configure the correct chain info. 

> Do we need to pass in secrets within the [deploy action](https://github.com/penumbra-zone/dex-explorer/blob/main/.github/workflows/deploy.yml)?

No, we don't: the environment variables are provided to the container at runtime, via configuration management software that isn't visible in this repo. I've confirmed that the `PENUMBRA_GRPC_ENDPOINT` var is available inside the container context:

```
/app $ printenv | grep ^PENUMBRA_GRPC_ENDPOINT
PENUMBRA_GRPC_ENDPOINT=https://testnet.plinfra.net
```

Indeed, all the necessary env vars are available and correct:

```
/app $ printenv | grep ^PENUMBRA_ | cut -d= -f1 | sort
PENUMBRA_CHAIN_ID
PENUMBRA_CUILOA_URL
PENUMBRA_GRPC_ENDPOINT
PENUMBRA_INDEXER_CA_CERT
PENUMBRA_INDEXER_ENDPOINT
```

So I don't think the problem is that the env vars are absent, I think the problem is that the `pnpm run build` process is (again) generating code that does not accept runtime configuration for the chain id. We need to fix that behavior, so that the app remains configurable. We'd initially addressed that behavior in #80, but due to recent code churn, compounded by the fact that the testnet deployment wasn't working well (#120), it's likely we broke that behavior without noticing the implications. 

What do you think, @grod220? Can we support the runtime config setup again?
